src/net-ipfs-engine/Catalyst.Ipfs.Http.Server/HttpApi/V0/KeyController.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Catalyst.Ipfs.Core.CoreApi;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Catalyst.Ipfs.Http.Server.HttpApi.V0
+{
+    /// <summary>
+    ///   A cryptographic key.
+    /// </summary>
+    public class CryptoKeyDto
+    {
+        /// <summary>
+        ///   The key's local name.
+        /// </summary>
+        public string Name;
+
+        /// <summary>
+        ///   The key's global unique ID.
+        /// </summary>
+        public string Id;
+    }
+
+    /// <summary>
+    ///   A list of cryptographic keys.
+    /// </summary>
+    public class CryptoKeysDto
+    {
+        /// <summary>
+        ///   A list of cryptographic keys.
+        /// </summary>
+        public IEnumerable<CryptoKeyDto> Keys;
+    }
+
+    /// <summary>
+    ///   A cryptographic key.
+    /// </summary>
+    public class CryptoKeyRenameDto
+    {
+        /// <summary>
+        ///   The key's local name.
+        /// </summary>
+        public string Was;
+
+        /// <summary>
+        ///   The key's global unique ID.
+        /// </summary>
+        public string Now;
+
+        /// <summary>
+        ///   The key's global unique ID.
+        /// </summary>
+        public string Id;
+
+        /// <summary>
+        ///   Indicates that a existing key was overwritten.
+        /// </summary>
+        public bool Overwrite;
+    }
+
+    /// <summary>
+    ///   Manages the cryptographic keys.
+    /// </summary>
+    public class KeyController : IpfsController
+    {
+        /// <summary>
+        ///   Creates a new controller.
+        /// </summary>
+        public KeyController(ICoreApi ipfs) : base(ipfs) { }
+
+        /// <summary>
+        ///   List all the keys.
+        /// </summary>
+        [HttpGet, HttpPost, Route("key/list")]
+        public async Task<CryptoKeysDto> List()
+        {
+            var keys = await IpfsCore.Key.ListAsync(Cancel);
+            return new CryptoKeysDto
+            {
+                Keys = keys.Select(key => new CryptoKeyDto
+                {
+                    Name = key.Name,
+                    Id = key.Id.ToString()
+                })
+            };
+        }
+
+        /// <summary>
+        ///   Create a new key.
+        /// </summary>
+        /// <param name="arg">
+        ///   The name of the key.
+        /// </param>
+        /// <param name="type">
+        ///   "rsa"
+        /// </param>
+        /// <param name="size">
+        ///   The key size in bits, if the type requires it.
+        /// </param>
+        [HttpGet, HttpPost, Route("key/gen")]
+        public async Task<CryptoKeyDto> Create(
+            string arg,
+            string type,
+            int size)
+        {
+            if (String.IsNullOrWhiteSpace(arg))
+                throw new ArgumentNullException("arg", "The key name is required.");
+            if (String.IsNullOrWhiteSpace(type))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)